### PR TITLE
Manage omnibus config api timeout error

### PR DESCRIPTION
**Pull request by vdeshkar**
- Added Timeout exception & error message

### CODE REVIEW

File: kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java [hunk 1]
```diff
@@ -84,16 +86,24 @@ public Map<String, Object> getConfig(ServiceRequest request, String environment,
 						} else {
 							LOG.warn("Invalid value type for JSON property: " + node.getNodeType().name());
 						}
+						}
 					}
+					else {
+						throw new TimeoutException("Timeout occurred when fetching configuration parameters");
+					}
+					CACHE.put(url, new CachedResult(retval));
+				} catch (Exception e) {
```
Comment (marianbuenosayres) [inline kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java:95]: Suggestion, adding a catch block here like this:
`catch (TimeoutException e) { throw e; }`
to not nest the exception unnecessarily

File: kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java [hunk 2]
```diff
@@ -84,16 +86,24 @@ public Map<String, Object> getConfig(ServiceRequest request, String environment,
...
+					LOG.warn("Error invoking config-api", e);
 				}
-				CACHE.put(url, new CachedResult(retval));
-			} catch (Exception e) {
```
Comment (marianbuenosayres) [inline kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java:90]: Suggestion, adding a catch block here like this:
`catch (TimeoutException e) { throw e; }`
to not nest the exception unnecessarily

File: kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java [hunk 3]
```diff
@@ -84,16 +86,24 @@ public Map<String, Object> getConfig(ServiceRequest request, String environment,
...
-				LOG.warn("Error invoking config-api", e);
 			}
-		}
+		
 		LOG.info("ConfigAPIUtil fetched {} variable values", CACHE.get(url).getVariables().size());
 		return retval;
-	}
+		}catch (Exception e) {
```
Comment (marianbuenosayres) [inline kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java:102]: Suggestion, adding a catch block here like this:
`catch (TimeoutException e) { throw e; }`
to not nest the exception unnecessarily

File: kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIProcessVariableInitHelper.java [hunk 4]
```diff
@@ -10,14 +11,18 @@
...
 		Map<String, Object> serviceProperty = super.initVariables(dep);
 		String env = System.getProperty("configApiEnv");
 		String appName = config.getProperty("kie.project.appname");
 		if (appName == null || "".equals(appName.trim())) {
 			appName = dep.getArtifactId();
 		}
-		serviceProperty.putAll(new ConfigAPIUtil().getConfig(request, env, client, appName));
+		try {
+			serviceProperty.putAll(new ConfigAPIUtil().getConfig(request, env, client, appName));
+		} catch (TimeoutException e) {
```
Comment (marianbuenosayres) [inline kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIProcessVariableInitHelper.java:23]: Suggestion, changing this catch block here to this:
`catch (TimeoutException e) { throw e; }`

**Review by marianbuenosayres — CHANGES_REQUESTED** [review]
I added a few comments about re-throwing existing TimeoutExceptions if possible. Also, I think we still need a special catch in GenericKieBasedService.execute(...) method for capturing the TimeoutException, and handle a specific ServiceResponse

**Comment by vdeshkar** [comment]
I can throw the exception the way you suggested, from where we will return the message, Is it handle at front end hide.?

**Pull request merged**